### PR TITLE
[receiver/zipkin] follow receiver contract

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** I came across `zipkinreceiver` and observed we don't follow the receiver [contract](https://github.com/open-telemetry/opentelemetry-collector/blob/b2961b799e2c1ec128f0539764af1fa10c839e04/receiver/doc.go#L21). We return `InternalServerError` straight away without checking permanent/non-permanent errors.

We should probably return BadRequest in case of permanent errors

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/4335

**Testing:** Added test cases

### CODE REVIEW

**Review by andrzej-stencel — APPROVED** [review]
Thanks a lot, I believe this is a step in the right direction.

**Comment by VihasMakwana** [comment]
> Thanks a lot, I believe this is a step in the right direction.

Thanks for your review, Andrzej.
I think it would also make sense to return the contents of the error as a part of the response. 
This can be a separate PR in itself. What do you think?

File: receiver/zipkinreceiver/trace_receiver.go [hunk 1]
```diff
@@ -239,17 +241,23 @@ func (zr *zipkinReceiver) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	if consumerErr == nil {
+		// Send back the response "Accepted" as
+		// required at https://zipkin.io/zipkin-api/#/default/post_spans
+		w.WriteHeader(http.StatusAccepted)
+		return
+	}
 
-	if consumerErr != nil {
+	if consumererror.IsPermanent(consumerErr) {
+		// Permanent error.
```
Comment (crobert-1) [inline receiver/zipkinreceiver/trace_receiver.go:252]: This comment isn't really helpful, as the conditional statement makes it clear that it's a permanent error. I'd say either explain why we're doing what we're doing here, or remove the comment.

**Review by crobert-1 — APPROVED** [review]
Thanks for making this change!

File: receiver/zipkinreceiver/trace_receiver.go (hunk 1 above)
Comment (VihasMakwana) [reply receiver/zipkinreceiver/trace_receiver.go]: Removing it for now.

**Comment by VihasMakwana** [comment]
@astencel-sumo @crobert-1 can we merge this one, it might be a good candidate for 0.89 release

**Review by codeboten — APPROVED** [review]

**Pull request merged**